### ISSUE TITLE
Intel graphical glitches when booting up laptop

### ISSUE THREAD

**Issue by Dycool** [holoiso-eol/holoiso#828]
## Your Issue:
```
I just installed holoISO and when it appears the screen to choose the language, the screen is filled with graphical glitches. I cant even see the languages.
```

## Your device's full specs
```
CPU: i5-1035G7
GPU: Intel Iris Plus Graphics
RAM: 16 GB
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
snapshot20230823.1454
```

## (Only for SteamOS Session loading issues) Have you tried launching `gamescope-session` from terminal? If yes, provide a log.
```
*Please put anything here if you tried to launch session separately*
```

## Have you updated your installation recently?
```
Its brand new.
```

**Comment by miaulightouch** [comment]
## For any one that want to run latest holoiso on intel igpu:
TLDR: **DON'T**
it's not supported, the hardware support is removed.

## Details

- intel drm driver from holoiso repo is removed
  - no iris, i915 driver found in /usr/lib/dri
- intel igpu require some config that already removed
  - env: `INTEL_DEBUG=noccs`
  - [optional] hybrid laptop requiregamescope parameter: `--prefer-vk-device=<your igpu id>`
- glitch in desktop mode
  - incompatible linux kernel, cause igpu keeping crash.

it's not meant that holoiso is bad, it just focus on better performance and other AMD variant handheld PC support. 

## Prepare
> tested on [ISO Update 5.0.1 (image snapshot20230823.1454)](https://github.com/HoloISO/holoiso/releases/tag/5.0.1)

I gonna use another pc to modify via ssh, so we need network connection.

- wired: just plug cable to laptop
- wireless:
  - press `Ctrl+Alt+2`
  - login to `root`
  - type: `nmcli dev wifi connect <wifi name> password <wifi password>`

find the ip of this laptop
- type: `ip addr`, and show like:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp2s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
3: wlo1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
    altname wlp0s20f3
    inet 192.168.1.100/24 metric 1024 brd 192.168.1.255 scope global dynamic noprefixroute wlo1
       valid_lft 591sec preferred_lft 591sec
    inet6 fe80::96c3:775a:54e7:5db3/64 scope link noprefixroute
       valid_lft forever preferred_lft forever
```

we can see our ip is `192.168.1.100`.

now, we execute from other pc to access: `ssh root@192.168.1.100`

## Fix Gamemode

do full system update first: `pacman -Syu`, we need newer holoiso installation to modify.

### Revert graphic driver

excute:
```sh
pacman -S extra-main/libva-mesa-driver extra-main/mesa-vdpau extra-main/opencl-mesa extra-main/vulkan-intel extra-main/vulkan-radeon multilib-main/lib32-libva-mesa-driver multilib-main/lib32-mesa multilib-main/lib32-mesa-vdpau multilib-main/lib32-vulkan-intel multilib-main/lib32-vulkan-radeon
```

### Add holoiso hardware profile for this laptop

#### glitch fix
```sh
echo export INTEL_DEBUG=noccs >> /usr/lib/holoiso-hwsupport/him_devicequirks/$(cat /sys/devices/virtual/dmi/id/product_name | tr -d '[:space:]').sh
````

reboot.

#### [Optional] hybrid gpu support

1. find igpu id: `lspci -knn`:
```
......
00:02.0 VGA compatible controller [0300]: Intel Corporation Iris Plus Graphics G1 (Ice Lake) [8086:8a56] (rev 07) (prog-if 00 [VGA controller])
......
03:00.0 Display controller [0380]: Advanced Micro Devices, Inc. [AMD/ATI] Topaz XT [Radeon R7 M260/M265 / M340/M360 / M440/M445 / 530/535 / 620/625 Mobile] [1002:6900] (rev d3)
```
`8086:8a56` is what we need.

2. add to hardware profile to ask gamemode launch on igpu
```sh
echo export HOLOISO_GAMESCOPE_ADDITIONAL_OPTIONS=\"--prefer-vk-device=8086:8a56\" >> /usr/lib/holoiso-hwsupport/him_devicequirks/$(cat /sys/devices/virtual/dmi/id/product_name | tr -d '[:space:]').sh
```

3. reboot

NOTE: run game on discrete gpu: modify game lauch option like this video: https://www.youtube.com/watch?v=9m5RzZ1P1g8

## Fix Desktop Mode
even we success to enter gamemode, but still glitch in desktop mode.

### Revert linux and firmware

execute:
```sh
pacman -S core-main/linux-firmware core-main/linux-firmware-marvell core-main/linux-firmware-whence extra-main/linux-zen extra-main/linux-zen-headers
```

reboot
